ENG-422 Create `swirl-toggle-button` and `swirl-toggle-group` components

matvlaovi-flip opened the pull request:
## Summary

- [Ticket](https://linear.app/flip/issue/ENG-422/implement-component-for-feeddetail-view-toggle)
- [Design](https://www.figma.com/design/rf6u4HqYxIk0n8SxcwylcU/Post-Preview?node-id=241-26279&t=zvF8Jsvoi0xjfNH5-4)
- [Storybook](https://icy-water-049ec4003-838.westeurope.3.azurestaticapps.net/?path=/docs/components-swirltogglegroup--docs) 

## Review Checklist

### General

- [x] [Changeset added](https://github.com/changesets/changesets/blob/main/docs/intro-to-using-changesets.md)
- [x] The submitted code is organized and formatted according to our [💅 Style Guide](https://swirl-storybook.flip-app.dev/?path=/docs/requirements-style-guide--docs).
  - [x] The code is formatted with Prettier.
  - [x] There are no linting errors.

### For new or updated components

https://swirl-storybook.flip-app.dev/?path=/docs/contributions-merge-publish--page

- [x] The changes do not contain any breaking changes.
- [x] The changes do not introduce new components that don't belong in the library (e.g. non-reusable components, highly specialized components, components including business logic)
- [x] The component documentation is updated.
- [x] The changes meet the [🤖 testing requirements](https://swirl-storybook.flip-app.dev/?path=/docs/requirements-testing--docs).
  - [x] New features are tested.
  - [ ] In case of bug fixes, regression tests have been added.
  - [x] All tests are 🟢.
- [x] The changes meet the [🧏‍♀️ accessibility requirements](https://swirl-storybook.flip-app.dev/?path=/docs/requirements-accessibility--docs).
  - [x] WCAG 2.1 Level A and Level AA requirements are met.
  - [x] The Storybook a11y addon shows no errors.
  - [ ] The changes have been tested with a screen reader.
  - [ ] Keyboard controls have been tested, if applicable.
  - [ ] Components implementing form controls (inputs, buttons, selects, etc.) do not use Shadow DOM, and instead use Stencil's scoping mechanism
- [ ] The changes use our [🌈 theming concept](https://swirl-storybook.flip-app.dev/?path=/docs/requirements-theming--docs).
  - [x] Design tokens have been used where appropriate.
  - [ ] The component has been visually checked in combination with the "Light" and "Dark" theme.
- [ ] The changes meet our [🌍 internationalization requirements](https://swirl-storybook.flip-app.dev/?path=/docs/requirements-internationalization--docs).
  - [ ] No static text is used.
  - [ ] The component doesn't break with longer texts or different text wrappings.
  - [ ] Number, currency and date values can be formatted appropriately.
- [ ] The changes work in all supported browsers and viewports. See [📱 Responsive Design](https://swirl-storybook.flip-app.dev/?path=/docs/requirements-responsive-design--docs)

Sqrrl commented on packages/swirl-components/src/components/swirl-toggle-button/swirl-toggle-button.css:
```diff
@@ -0,0 +1,54 @@
+.button {
+  display: inline-flex;
+  width: 100%;
+  padding: var(--s-space-8) var(--s-space-16);
```
Issue: Horizontal padding seems to be 12 in Figma.

Sqrrl commented on packages/swirl-components/src/components/swirl-toggle-button/swirl-toggle-button.css:
```diff
@@ -0,0 +1,54 @@
+  justify-content: center;
+  align-items: center;
+  border: none;
+  border-radius: var(--s-border-radius-base);
```
Issue: Figma says `border-radius-sm`. Hope I'm looking at the right designs.

Sqrrl commented on packages/swirl-components/src/components/swirl-toggle-button/swirl-toggle-button.css:
```diff
@@ -0,0 +1,54 @@
+  border-radius: var(--s-border-radius-base);
+  color: var(--s-text-default);
+  background-color: var(--swirl-ghost-button-background-default);
+  font: inherit;
```
Issue: Font size should be `font-size-sm`.

Sqrrl commented on packages/swirl-components/src/components/swirl-toggle-button/swirl-toggle-button.css:
```diff
@@ -0,0 +1,54 @@
+  background-color: var(--swirl-ghost-button-background-default);
+  font: inherit;
+  font-weight: var(--s-font-weight-medium);
+  line-height: var(--s-line-height-lg);
```
Issue: `line-height-sm`

Sqrrl commented on packages/swirl-components/src/components/swirl-toggle-button/swirl-toggle-button.css:
```diff
@@ -0,0 +1,54 @@
+  text-decoration: none;
+  cursor: pointer;
+  gap: var(--s-space-8);
+
```
Suggestion: Could be a good idea to prevent it from wrapping and add ellipsis truncation.

Sqrrl commented on packages/swirl-components/src/components/swirl-toggle-button/swirl-toggle-button.tsx:
```diff
@@ -0,0 +1,33 @@
+        class={classNames}
+        aria-pressed={this.isPressed ? "true" : "false"}
+        onClick={this.onClick}
+      >
```
Suggestion: `type="button"` so it doesn't accidentally submits a form on the page.

Sqrrl commented:
Just noticed that you probably used the mobile swirl-button styles for the toggle buttons here, which is fine. You just need to also add the desktop media query overwrites: https://github.com/getflip/swirl/blob/7df651b877b6a902c8bacdf18cba95cb727f9c59/packages/swirl-components/src/components/swirl-button/swirl-button.css#L62

matvlaovi-flip commented:
> Just noticed that you probably used the mobile swirl-button styles for the toggle buttons here, which is fine. You just need to also add the desktop media query overwrites:
> 
> https://github.com/getflip/swirl/blob/7df651b877b6a902c8bacdf18cba95cb727f9c59/packages/swirl-components/src/components/swirl-button/swirl-button.css#L62
> 
> ```css
> @media (--from-desktop-without-touch) {
>     padding: var(--s-space-8) var(--s-space-12);
>     font-size: var(--s-font-size-sm);
>     line-height: var(--s-line-height-sm);
>     border-radius: var(--s-border-radius-sm);
> }
> ```

Good catch! Adjusted and added the truncation. Border radius will now be handled with the radius variable, just as swirl-button. Also removed the `gap` as this one will only have a label inside
